### PR TITLE
python310Packages.vector: adjust inputs

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Fix build (https://hydra.nixos.org/build/205483689)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 211929` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>av-98</li>
    <li>jrnl</li>
    <li>python310Packages.ansiwrap</li>
    <li>python310Packages.papermill</li>
    <li>python310Packages.vector</li>
    <li>python311Packages.ansiwrap</li>
    <li>python311Packages.papermill</li>
  </ul>
</details>

File: pkgs/development/python-modules/vector/default.nix [hunk 1]
```diff
@@ -21,17 +27,31 @@ buildPythonPackage rec {
...
   ];
 
-  checkInputs = [ pytestCheckHook ];
+  pythonImportsCheck = [
+    "vector"
+  ];
 
-  pythonImportsCheck = [ "vector" ];
+  disabledTestPaths = [
+    # jupyter_client.kernelspec.NoSuchKernel: No such kernel named python3
```
Comment (veprbl) [inline pkgs/development/python-modules/vector/default.nix:47]: Sinc you add the dependency on `papermill`, might as well fix the test:
```diff
diff --git a/pkgs/development/python-modules/vector/default.nix b/pkgs/development/python-modules/vector/default.nix
index e01394a5637..1c8a7d4d86d 100644
--- a/pkgs/development/python-modules/vector/default.nix
+++ b/pkgs/development/python-modules/vector/default.nix
@@ -4,9 +4,11 @@
 , fetchPypi
 , hatch-vcs
 , hatchling
+, numba
 , numpy
 , packaging
 , papermill
+, notebook
 , pytestCheckHook
 , pythonOlder
 }:
@@ -36,6 +38,8 @@ buildPythonPackage rec {
   checkInputs = [
     awkward
     papermill
+    numba
+    notebook
     pytestCheckHook
   ];
 
@@ -43,10 +47,7 @@ buildPythonPackage rec {
     "vector"
   ];
 
-  disabledTestPaths = [
-    # jupyter_client.kernelspec.NoSuchKernel: No such kernel named python3
-    "tests/test_notebooks.py"
-  ];
+  __darwinAllowLocalNetworking = true;
 
   meta = with lib; {
     description = "Library for 2D, 3D, and Lorentz vectors, especially arrays of vectors, to solve common physics problems in a NumPy-like way";
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 211929` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>av-98</li>
    <li>jrnl</li>
    <li>python310Packages.ansiwrap</li>
    <li>python310Packages.papermill</li>
    <li>python310Packages.vector</li>
    <li>python311Packages.ansiwrap</li>
    <li>python311Packages.papermill</li>
  </ul>
</details>

File: pkgs/development/python-modules/vector/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/vector/default.nix]: Thanks, added.

**Review by veprbl — APPROVED** [review]
LGTM

**Pull request merged**